lightly/api/download.py

```diff
@@ -154,7 +170,7 @@ def video_frame_count(
             decoded.
 
         """
-        with av.open(url) as container:
+        with retry_fn(av.open, url) as container:
```

Comment:
Here I would retry the whole av.open block instead of just the open itself. Alternatively, do not retry at all. The actual frames are only loaded when the container is decoded. Retrying only the open, but not the frame loading does not make sense.

Reply:
Ah, I maybe should have explained this better. In general we always use two nested retries when reading images/videos. The inner retry handles the opening part of the file/video. The outer retry is responsible for handling any errors that happen while the file is downloaded. For example if you try to load a file from an invalid url the inner retry will fail. If you have internet connection problems while loading a huge video file then it will probably fail in the outer retry.

The reason why there is not only an outer retry is two-fold. First, some libraries like `requests` already partially handle retries themselves, so you get automatically an inner retry. Second, the two retries handle different error cases and they should probably not interfere with each other. Note that if the inner retry fails because it reaches the maximum number of retries then the outer retry will NOT retry again.

Reply:
Thanks for explaining this well :)